src/Certificate/symmetry.jl

```diff
@@ -0,0 +1,114 @@
+#import GroupsCore
+import SymbolicWedderburn
+#import PermutationGroups
+#import Cyclotomics
+
+function SymbolicWedderburn.decompose(
+    k::MP.AbstractPolynomialLike,
+    hom::SymbolicWedderburn.InducedActionHomomorphism,
+)
+    # correct only if features(hom) == monomials
+
+    indcs = [hom[mono] for mono in MP.monomials(k)]
+    coeffs = MP.coefficients(k)
+
+    return indcs, coeffs
+end
+
+function SymbolicWedderburn.ExtensionHomomorphism(
+    action::SymbolicWedderburn.Action,
+    basis::MB.MonomialBasis,
+)
+    monos = collect(basis.monomials)
+    mono_to_index = Dict(monos[i] => i for i in eachindex(monos))
+    return SymbolicWedderburn.ExtensionHomomorphism(action, monos, mono_to_index)
+end
+
+struct VariablePermutation <: SymbolicWedderburn.ByPermutations end
+_map_idx(f, v::AbstractVector) = map(f, eachindex(v))
+_tuple_map_idx(f, ::Tuple{}, i) = tuple()
+_tuple_map_idx(f, v::Tuple, i) = (f(i), _tuple_map_idx(f, Base.tail(v), i + 1)...)
+_map_idx(f, v::Tuple) = _tuple_map_idx(f, v, 1)
+function SymbolicWedderburn.action(::VariablePermutation, p, mono::MP.AbstractMonomial)
+    v = variables(mono)
+    MP.substitute(MP.Eval(), mono, v => _map_idx(i -> v[i^p], v))
+end
+abstract type MonomialTransformation <: SymbolicWedderburn.ByLinearTransformation end
```

Comment:
@kalmarek What do you think of the name ? `MonomialAction` would be better ?

Reply:
`OnMonomials`? (then you can have `action(::OnMonomials, el, term)`, which seems quite natural) but all names are fine to me

Reply:
Aaah that's why you call them like that, the subtyping also looks good, e.g., `DihedralAction <: OnMonomials` reads like "Dihedral action on monomials"